submit egs2/tedlium2/asr1

flylili001 opened the pull request:
please check, I finish the egs of tedlium2 on espnet2.

flylili001 commented:
please check egs2/tedlium2/asr1

sw005320 commented:
Thanks!
@ftshijt, could you review this PR?

ftshijt commented on egs2/tedlium2/asr1/asr.sh:
```diff
@@ -0,0 +1,1348 @@
+#!/bin/bash
```
Please use soft link (e.g. `ln -s`) and refer files in TEMPLATE for both asr.sh and db.sh

ftshijt commented on egs2/tedlium2/asr1/conf/train_asr_transformer.yaml:
```diff
@@ -0,0 +1,62 @@
+batch_type: numel
```
I found some differences between this config and that in `conf/tuning`. Which one did you use for the RESULT? We could just keep them in the `tuning` and use soft-link in conf.

ftshijt commented on egs2/tedlium2/asr1/local/data.sh:
```diff
@@ -0,0 +1,29 @@
+# general configuration
+stage=-1       # start from -1 if you need to start from data download
+stop_stage=100
+
```
please add `. utils/parse_options.sh || exit 1;` to allow argument parsing here.

ftshijt commented on egs2/tedlium2/asr1/local/download_data.sh:
```diff
@@ -0,0 +1,38 @@
+# TED-LIUM database:
+if [[ $(hostname -f) == *.clsp.jhu.edu ]] ; then
+  if [ ! -e TEDLIUM_release2 ]; then
+    ln -sf /export/corpora5/TEDLIUM_release2
```
Please reflect this in `db.sh` instead of here.

ftshijt reviewed (commented):
Looks great! I just have some minor comments to make it standardized to other recipes.

flylili001 commented:
> Please use soft link (e.g. ln -s) and refer files in TEMPLATE for both asr.sh and db.sh
Please help me to change it for soft link, I don't know how to make a soft link in windows github.


> I found some differences between this config and that in conf/tuning. Which one did you use for the RESULT? We could just keep them in the tuning and use soft-link in conf.
I used the configure of conf/train_asr_transformer.yaml. Please copy it in conf/tuning and make a soft-link.


> please add . utils/parse_options.sh || exit 1; to allow argument parsing here.
OK. Please add it in local/data.sh


> Please reflect this in db.sh instead of here.
OK. Please help me to add it in db.sh. I think db.sh is the file in TEMPLATE, and I can't modify it.




Thank you.










At 2021-01-01 04:51:17, "ftshijt" <notifications@github.com> wrote:

@ftshijt commented on this pull request.

Looks great! I just have some minor comments to make it standardized to other recipes.

In egs2/tedlium2/asr1/asr.sh:

> @@ -0,0 +1,1348 @@
+#!/bin/bash


Please use soft link (e.g. ln -s) and refer files in TEMPLATE for both asr.sh and db.sh

In egs2/tedlium2/asr1/conf/train_asr_transformer.yaml:

> @@ -0,0 +1,62 @@
+batch_type: numel


I found some differences between this config and that in conf/tuning. Which one did you use for the RESULT? We could just keep them in the tuning and use soft-link in conf.

In egs2/tedlium2/asr1/local/data.sh:

> @@ -0,0 +1,29 @@
+#!/bin/bash
+
+# Copyright 2017 Johns Hopkins University (Shinji Watanabe)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+# Modified by flylili001
+
+. ./path.sh || exit 1;
+. ./cmd.sh || exit 1;
+
+# general configuration
+stage=-1       # start from -1 if you need to start from data download
+stop_stage=100
+


please add . utils/parse_options.sh || exit 1; to allow argument parsing here.

In egs2/tedlium2/asr1/local/download_data.sh:

> @@ -0,0 +1,38 @@
+#!/bin/bash
+
+# Copyright  2014  Nickolay V. Shmyrev
+#            2014  Brno University of Technology (Author: Karel Vesely)
+#            2016  John Hopkins University (author: Daniel Povey)
+# Apache 2.0
+
+mkdir -p db
+
+cd db  ### Note: the rest of this script is executed from the directory 'db'.
+
+# TED-LIUM database:
+if [[ $(hostname -f) == *.clsp.jhu.edu ]] ; then
+  if [ ! -e TEDLIUM_release2 ]; then
+    ln -sf /export/corpora5/TEDLIUM_release2


Please reflect this in db.sh instead of here.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

ftshijt commented:
> Please use soft link (e.g. ln -s) and refer files in TEMPLATE for both asr.sh and db.sh
> Please help me to change it for soft link, I don't know how to make a soft link in windows github.
> I found some differences between this config and that in conf/tuning. Which one did you use for the RESULT? We could just keep them in the tuning and use soft-link in conf.
> I used the configure of conf/train_asr_transformer.yaml. Please copy it in conf/tuning and make a soft-link.
> please add . utils/parse_options.sh || exit 1; to allow argument parsing here.
> OK. Please add it in local/data.sh
> Please reflect this in db.sh instead of here.
> OK. Please help me to add it in db.sh. I think db.sh is the file in TEMPLATE, and I can't modify it. Thank you. At 2021-01-01 04:51:17, "ftshijt" <notifications@github.com> wrote: @ftshijt commented on this pull request. Looks great! I just have some minor comments to make it standardized to other recipes. In egs2/tedlium2/asr1/asr.sh:
> @@ -0,0 +1,1348 @@
> +#!/bin/bash Please use soft link (e.g. ln -s) and refer files in TEMPLATE for both asr.sh and db.sh In egs2/tedlium2/asr1/conf/train_asr_transformer.yaml:
> @@ -0,0 +1,62 @@
> +batch_type: numel I found some differences between this config and that in conf/tuning. Which one did you use for the RESULT? We could just keep them in the tuning and use soft-link in conf. In egs2/tedlium2/asr1/local/data.sh:
> @@ -0,0 +1,29 @@
> +#!/bin/bash + +# Copyright 2017 Johns Hopkins University (Shinji Watanabe) +# Apache 2.0 (http://www.apache.org/licenses/LICENSE-2.0) +# Modified by flylili001 + +. ./path.sh || exit 1; +. ./cmd.sh || exit 1; + +# general configuration +stage=-1 # start from -1 if you need to start from data download +stop_stage=100 + please add . utils/parse_options.sh || exit 1; to allow argument parsing here. In egs2/tedlium2/asr1/local/download_data.sh:
> @@ -0,0 +1,38 @@
> +#!/bin/bash + +# Copyright 2014 Nickolay V. Shmyrev +# 2014 Brno University of Technology (Author: Karel Vesely) +# 2016 John Hopkins University (author: Daniel Povey) +# Apache 2.0 + +mkdir -p db + +cd db ### Note: the rest of this script is executed from the directory 'db'. + +# TED-LIUM database: +if [[ $(hostname -f) == *.clsp.jhu.edu ]] ; then + if [ ! -e TEDLIUM_release2 ]; then + ln -sf /export/corpora5/TEDLIUM_release2 Please reflect this in db.sh instead of here. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

Hi, The soft-link could be add using `ln -s <source_path> <target_path>` and them simply use `git add` and `commit` to add them. I could take over it if it is not easy for you to do, but could you manage to pass the CI tests first? That would be much helpful.

flylili001 commented:
Please help me to modify the soft-links, and could you manage to pass the CI tests? If there is any issues, please tell me.
Thank you.




bests,
Lili










At 2021-01-05 09:57:11, "ftshijt" <notifications@github.com> wrote:

Please use soft link (e.g. ln -s) and refer files in TEMPLATE for both asr.sh and db.sh
Please help me to change it for soft link, I don't know how to make a soft link in windows github.
I found some differences between this config and that in conf/tuning. Which one did you use for the RESULT? We could just keep them in the tuning and use soft-link in conf.
I used the configure of conf/train_asr_transformer.yaml. Please copy it in conf/tuning and make a soft-link.
please add . utils/parse_options.sh || exit 1; to allow argument parsing here.
OK. Please add it in local/data.sh
Please reflect this in db.sh instead of here.
OK. Please help me to add it in db.sh. I think db.sh is the file in TEMPLATE, and I can't modify it. Thank you. At 2021-01-01 04:51:17, "ftshijt" notifications@github.com wrote: @ftshijt commented on this pull request. Looks great! I just have some minor comments to make it standardized to other recipes. In egs2/tedlium2/asr1/asr.sh:
@@ -0,0 +1,1348 @@
+#!/bin/bash Please use soft link (e.g. ln -s) and refer files in TEMPLATE for both asr.sh and db.sh In egs2/tedlium2/asr1/conf/train_asr_transformer.yaml:
@@ -0,0 +1,62 @@
+batch_type: numel I found some differences between this config and that in conf/tuning. Which one did you use for the RESULT? We could just keep them in the tuning and use soft-link in conf. In egs2/tedlium2/asr1/local/data.sh:
@@ -0,0 +1,29 @@
+#!/bin/bash + +# Copyright 2017 Johns Hopkins University (Shinji Watanabe) +# Apache 2.0 (http://www.apache.org/licenses/LICENSE-2.0) +# Modified by flylili001 + +. ./path.sh || exit 1; +. ./cmd.sh || exit 1; + +# general configuration +stage=-1 # start from -1 if you need to start from data download +stop_stage=100 + please add . utils/parse_options.sh || exit 1; to allow argument parsing here. In egs2/tedlium2/asr1/local/download_data.sh:
@@ -0,0 +1,38 @@
+#!/bin/bash + +# Copyright 2014 Nickolay V. Shmyrev +# 2014 Brno University of Technology (Author: Karel Vesely) +# 2016 John Hopkins University (author: Daniel Povey) +# Apache 2.0 + +mkdir -p db + +cd db ### Note: the rest of this script is executed from the directory 'db'. + +# TED-LIUM database: +if [[ $(hostname -f) == *.clsp.jhu.edu ]] ; then + if [ ! -e TEDLIUM_release2 ]; then + ln -sf /export/corpora5/TEDLIUM_release2 Please reflect this in db.sh instead of here. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

Hi, The soft-link could be add using ln -s <source_path> <target_path> and them simply use git add and commit to add them. I could take over it if it is not easy for you to do, but could you manage to pass the CI tests first? That would be much helpful.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

ftshijt commented:
@flylili001 Hi, could you merge my PR to your repo so as to reflect some minor changes to the code (i.e., soft link and structural stuff)?

ftshijt commented:
@flylili001 Hi, we recently updated the contributing process, where we suggested several tips for you to prepare the recipe. Please check this out: https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md

flylili001 commented:
Hi, I checked the website. And I think Zenodo tool is only for model upload, but not for recipe. Sorry, I don't know how to upload the recipe of tedlium2 with Zenodo.










At 2021-02-02 02:52:33, "ftshijt" <notifications@github.com> wrote:

@flylili001 Hi, we recently updated the contributing process, where we suggested several tips for you to prepare the recipe. Please check this out: https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

ftshijt commented:
> Hi, I checked the website. And I think Zenodo tool is only for model upload, but not for recipe. Sorry, I don't know how to upload the recipe of tedlium2 with Zenodo. At 2021-02-02 02:52:33, "ftshijt" <notifications@github.com> wrote: @flylili001 Hi, we recently updated the contributing process, where we suggested several tips for you to prepare the recipe. Please check this out: https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

Yes, the zenodo is for model uploading. You are not required to upload the model but are strongly recommended. Currently, what concerns most is that you did not follow the requirements discussed in 1.3.3 (https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md). In the case that you are not able to set up those, I've started a PR to your repo (see https://github.com/flylili001/espnet/pull/1) Please check and merge that to reflect the changes. Thanks.

flylili001 commented:
Hi, I have modified all issues. Please check it. https://github.com/flylili001/espnet/


Thanks!


Lili




At 2021-02-22 09:39:38, "ftshijt" <notifications@github.com> wrote:

Hi, I checked the website. And I think Zenodo tool is only for model upload, but not for recipe. Sorry, I don't know how to upload the recipe of tedlium2 with Zenodo. At 2021-02-02 02:52:33, "ftshijt" notifications@github.com wrote: @flylili001 Hi, we recently updated the contributing process, where we suggested several tips for you to prepare the recipe. Please check this out: https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

Yes, the zenodo is for model uploading. You are not required to upload the model but are strongly recommended. Currently, what concerns most is that you did not follow the requirements discussed in 1.3.3 (https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md). In the case that you are not able to set up those, I've started a PR to your repo (see flylili001#1) Please check and merge that to reflect the changes. Thanks.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

kamo-naoyuki commented on egs2/tedlium2/asr1/run.sh:
```diff
@@ -0,0 +1,31 @@
+    --ngpu 8 \
+    --nbpe 500 \
+    --max_wav_duration 20 \
+    --feats_type fbank_pitch \
```
It's not allow to  fbank_pitch for all recipe. Please use raw only.

kamo-naoyuki commented on egs2/tedlium2/asr1/local/data.sh:
```diff
@@ -0,0 +1,31 @@
+# Copyright 2017 Johns Hopkins University (Shinji Watanabe)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+# Modified by flylili001
+
```
Don't forget `set -euo pipefail`. This is very dangerous.

kamo-naoyuki commented on egs2/tedlium2/asr1/local/data.sh:
```diff
@@ -0,0 +1,31 @@
+. utils/parse_options.sh || exit 1;
+
+if [ ${stage} -le -1 ] && [ ${stop_stage} -ge -1 ]; then
+    echo "stage -1: Data Download"
```
Don't use `echo` for  logging purpose. Use log command.

kamo-naoyuki commented on egs2/tedlium2/asr1/local/data.sh:
```diff
@@ -0,0 +1,31 @@
+    echo "stage 0: Data preparation"
+    local/prepare_data.sh
+    for dset in dev test train; do
+    utils/data/modify_speaker_info.sh --seconds-per-spk-max 180 data/${dset}.orig data/${dset}
```
You can't use this script becasue it requires Kaldi. Remove this block.

kamo-naoyuki commented on egs2/tedlium2/asr1/path.sh:
```diff
@@ -0,0 +1,22 @@
+MAIN_ROOT=$PWD/../../..
+KALDI_ROOT=$MAIN_ROOT/tools/kaldi
```
Change this file to a symlink

`ln -s ../../TEMPLATE/asr1/path.sh`

kamo-naoyuki commented on egs2/tedlium2/asr1/run.sh:
```diff
@@ -0,0 +1,31 @@
+    --train_set "${train_set}" \
+    --valid_set "${valid_set}" \
+    --test_sets "${test_sets}" \
+    --srctexts "data/${train_set}/text" "$@"
```
Please update espnet to latest before creating PR. 

`--srctexts` doesn't exist.

flylili001 commented:
Thank you for your reviews.


I have modified all issues, please check again.







At 2021-02-25 18:49:17, "Naoyuki Kamo" <notifications@github.com> wrote:

@kamo-naoyuki commented on this pull request.

In egs2/tedlium2/asr1/local/data.sh:

> @@ -0,0 +1,31 @@
+#!/bin/bash
+
+# Copyright 2017 Johns Hopkins University (Shinji Watanabe)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+# Modified by flylili001
+


Don't forget set -euo pipefail. This is very dangerous.

In egs2/tedlium2/asr1/local/data.sh:

> +
+# Copyright 2017 Johns Hopkins University (Shinji Watanabe)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+# Modified by flylili001
+
+. ./path.sh || exit 1;
+. ./cmd.sh || exit 1;
+
+# general configuration
+stage=-1       # start from -1 if you need to start from data download
+stop_stage=100
+
+. utils/parse_options.sh || exit 1;
+
+if [ ${stage} -le -1 ] && [ ${stop_stage} -ge -1 ]; then
+    echo "stage -1: Data Download"


Don't use echo for logging purpose. Use log command.

In egs2/tedlium2/asr1/local/data.sh:

> +stop_stage=100
+
+. utils/parse_options.sh || exit 1;
+
+if [ ${stage} -le -1 ] && [ ${stop_stage} -ge -1 ]; then
+    echo "stage -1: Data Download"
+    local/download_data.sh
+fi
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    ### Task dependent. You have to make data the following preparation part by yourself.
+    ### But you can utilize Kaldi recipes in most cases
+    echo "stage 0: Data preparation"
+    local/prepare_data.sh
+    for dset in dev test train; do
+    utils/data/modify_speaker_info.sh --seconds-per-spk-max 180 data/${dset}.orig data/${dset}


You can't use this script becasue it requires Kaldi. Remove this block.

In egs2/tedlium2/asr1/path.sh:

> @@ -0,0 +1,22 @@
+MAIN_ROOT=$PWD/../../..
+KALDI_ROOT=$MAIN_ROOT/tools/kaldi


Change this file to a symlink

ln -s ../../TEMPLATE/asr1/path.sh

In egs2/tedlium2/asr1/run.sh:

> +stage=10
+./asr.sh \
+    --stage "${stage}" \
+    --lang en \
+    --ngpu 8 \
+    --nbpe 500 \
+    --max_wav_duration 20 \
+    --feats_type fbank_pitch \
+    --speed_perturb_factors "0.9 1.0 1.1" \
+    --asr_config "${asr_config}" \
+    --lm_config "${lm_config}" \
+    --inference_config "${inference_config}" \
+    --train_set "${train_set}" \
+    --valid_set "${valid_set}" \
+    --test_sets "${test_sets}" \
+    --srctexts "data/${train_set}/text" "$@"


Please update espnet to latest before creating PR.

--srctexts doesn't exist.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

ftshijt commented:
Hi, seems that you forgot to PR? We could not see your updates on git.

flylili001 commented:
Hi, I have updated README.md and added the PR at the end.

Please check it.
Thanks.







At 2021-02-26 21:15:03, "ftshijt" <notifications@github.com> wrote:

Hi, seems that you forgot to PR? We could not see your updates on git.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

ftshijt commented:
> Hi, I have updated README.md and added the PR at the end. Please check it. Thanks. At 2021-02-26 21:15:03, "ftshijt" <notifications@github.com> wrote: Hi, seems that you forgot to PR? We could not see your updates on git. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

Hi, thanks for reaching out. We are not able to see your modification on our end. Have you pushed your changes to your fork?

flylili001 commented:
Yes, I just update the code in my fork and I don't know how to push my project to your fork. Because I don't have permissions.
What should I do, please tell me.




Thanks.







At 2021-03-03 08:01:44, "ftshijt" <notifications@github.com> wrote:

Hi, I have updated README.md and added the PR at the end. Please check it. Thanks. At 2021-02-26 21:15:03, "ftshijt" notifications@github.com wrote: Hi, seems that you forgot to PR? We could not see your updates on git. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

Hi, thanks for reaching out. We are not able to see your modification on our end. Have you pushed your changes to your fork?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

ftshijt commented:
> Yes, I just update the code in my fork and I don't know how to push my project to your fork. Because I don't have permissions. What should I do, please tell me. Thanks. At 2021-03-03 08:01:44, "ftshijt" <notifications@github.com> wrote: Hi, I have updated README.md and added the PR at the end. Please check it. Thanks. At 2021-02-26 21:15:03, "ftshijt" notifications@github.com wrote: Hi, seems that you forgot to PR? We could not see your updates on git. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe. Hi, thanks for reaching out. We are not able to see your modification on our end. Have you pushed your changes to your fork? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

You simply need to push your results to your own fork on the git, then the results would sync to us. Please check these for reference: 

http://stackoverflow.com/questions/14680711/how-to-do-a-github-pull-request
https://help.github.com/articles/creating-a-pull-request/

flylili001 commented:
Yes, I have pushed the results to my git : https://github.com/flylili001/espnet/tree/master/egs2/tedlium2/asr1, please check it.
Any question, please tell me.

Thanks.













At 2021-03-05 22:33:05, "ftshijt" <notifications@github.com> wrote:

Yes, I just update the code in my fork and I don't know how to push my project to your fork. Because I don't have permissions. What should I do, please tell me. Thanks. At 2021-03-03 08:01:44, "ftshijt" notifications@github.com wrote: Hi, I have updated README.md and added the PR at the end. Please check it. Thanks. At 2021-02-26 21:15:03, "ftshijt" notifications@github.com wrote: Hi, seems that you forgot to PR? We could not see your updates on git. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe. Hi, thanks for reaching out. We are not able to see your modification on our end. Have you pushed your changes to your fork? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

You simply need to push your results to your own fork on the git, then the results would sync to us. Please check these for reference:

http://stackoverflow.com/questions/14680711/how-to-do-a-github-pull-request
https://help.github.com/articles/creating-a-pull-request/

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

flylili001 commented:
Sorry, I forgot to push. Please check again.
















At 2021-03-09 11:33:40, "ftshijt" <notifications@github.com> wrote:

https://github.com/flylili001/espnet/tree/master/egs2/tedlium2/asr1

Hi, seems that you still did not update the repo on the github. Maybe you are doing those locally?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

ftshijt commented:
I think we already got tedlium2 from #3795, so I will close this PR because of the conflict. Thanks for your effort @flylili001
